src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,73 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 2;
```

Comment:
The threshold value of 2 for switching to insertion sort is quite low. Typically, a higher threshold (like 10 or 20) is used to optimize performance, as insertion sort is more efficient for small arrays.